base/operators.jl

```diff
@@ -972,3 +972,22 @@ julia> filter(!isalpha, str)
 ```
 """
 !(f::Function) = (x...)->!f(x...)
+
+struct equalto{T} <: Function
```

Comment:
This should probably use the typical upper camel case style for type names as recommended in the style guide and used in Base, especially since this is user-facing.

Reply:
Disagree, since this is meant to be used like a function, `equalto(x)`.

Reply:
You can always define `equalto(x) = EqualTo(x)`, as is done with `Iterators.filter` returning a `Filter` object.

Reply:
A problem with conflating `equalto` with "regular" functions is that it doesn't have its own type in the same way that other functions do. In particular, to specialize on a particular function, you use `::typeof(f)`, but in this case `typeof(equalto)` will be `DataType`. So users will have to remember that to specialize on this function they need `::equalto` instead.

Reply:
I agree with @ararslan, that pattern is quite common, we also use it with `zip` vs. `Zip1`/`Zip2`.

Reply:
True, but I'm not sure that's such a great pattern. It's confusing to have both `filter` and `Filter`. That came from a situation where `filter` had several methods, only one of which returned a `Filter`, but then the functions were separated (into `Base.filter` and `Base.Iterators.filter`). `zip` is also different in that it might return either a `Zip1` or a `Zip2`, so it can't be directly identified with either type.